PSKoans/Koans/Cmdlets 1/AboutPSProviders.Koans.ps1

```diff
@@ -127,17 +127,33 @@ Describe 'Environment Provider' {
 }
 
 Describe 'FileSystem Provider' {
-    $Path = 'TestDrive:' | Join-Path -ChildPath 'File001.tmp'
+    if( $PSVersionTable.PSVersion.Major -le 5 ){
+        #If on windows 5 or earlier, use env temp directory.
+        $Path = Join-Path -Path $env:TEMP -ChildPath 'PSKoans.tmp'
+    }
+    elseif( (Get-Variable -Name IsWindows)){
+        if( $IsWindows){
+            $Path = Join-Path -Path $env:TEMP -ChildPath 'PSKoans.tmp'
+        }
+        elseif( (Test-Path \tmp)){
+            #We found a \tmp directory, let's use it!
+            $Path = Join-Path -Path \tmp -ChildPath 'PSKoans.tmp'
+        }
+        else{
+            #Path to MacOS temp directory
+            $Path = Join-Path -Path $env:TMPDIR -ChildPath 'PSKoans.tmp'
+        }
+    }
+    else{
+        Write-Error -Message "Was not able to determine temp directory location."
+    }
```

Comment:
Couple things here.

If `\tmp` is meant to be for Linux/Mac FS, you need to use forward slashes; their filesystem won't work with backslashes. 

Also, this is too complicated, especially for something we're setting up in a koan topic file. If we really need this boilerplate, I'd prefer it gets put in the PSM1 as a script-scoped variable and access to that data provided via a public module function (e.g., `Get-TempDirectory` or some such).

Relatedly, I don't want to be writing non-Pester-related errors from a koan lesson file. All it's going to do is confuse the user, unfortunately.

Reply:
1. Done, nice catch!
2. I started this off as a private function, and suppressed any errors. Let me know if you would like further changes.